src/NHibernate/Engine/CollectionEntry.cs

```diff
@@ -302,6 +302,10 @@ public void PostInitialize(IPersistentCollection collection)
 		{
 			snapshot = LoadedPersister.IsMutable ? collection.GetSnapshot(LoadedPersister) : null;
 			collection.SetSnapshot(loadedKey, role, snapshot);
+			if (LoadedPersister.GetBatchSize() > 1)
+			{
+				collection.GetCurrentSession().PersistenceContext.BatchFetchQueue.RemoveBatchLoadableCollection(this);
```

Comment:
It is no more a binary breaking change but it is still a breaking one, isn't it? Because if anyone supplies its own `IPersistentCollection` not deriving from `AbstractPersistentCollection`, it will now throw here.

Reply:
What if returning 0/1 on get batch size if this is not supported one?